### PR TITLE
Add an option for "no delay" in tab preview timing

### LINKED ISSUE #8860: Make tab preview based on mouse idle time
## Test plan
See https://github.com/brave/browser-laptop/pull/9887

After #12823 please add the NO DELAY (current name) as a check too for undelayed tab previews. Thanks!!

<hr>

## Original issue description
Sometimes tab preview happen unintentionally if you go from urlbar to webview. Tab previews are based on RegExp that checks which element has hovered before current tab preview. 

After taking #3271 this still seems to be the best approach currently but we need to cover other elements that were missing on initial implementation.

### PR DESCRIPTION
**Pull request by cezaraugusto**
closes #8860

this is a follow-up of #9887 and adds a _none_ option for tab preview delay.

Test Plan:

1. Go to pref->tabs
2. Confirm tab preview defaults to "short" 
3. Change tab preview timing to "none"
4. tab previews should be instant

### CODE REVIEW

File: app/renderer/components/tabs/tabs.js [hunk 1]
```diff
@@ -195,6 +196,7 @@ class Tabs extends React.Component {
                 key={'tab-' + frameKey}
                 ref={(node) => this.tabRefs.push(node)}
                 frameKey={frameKey}
+                tabsPerTabPage={this.props.tabsPerTabPage}
```
Comment (petemill) [inline app/renderer/components/tabs/tabs.js]: Wondering why a new prop has been added here, but no change to the Tab component to utilize it. Is this fixing a different bug?

Comment (cezaraugusto) [reply app/renderer/components/tabs/tabs.js]: oh my bad this was for another pr thanks for spotting

**Comment by cezaraugusto** [comment]
@petemill ya I like _instant_ and also suggested _immediate_ but I guess @bradleyrichter is searching for a better phrase for the select title.

re timing I'm ok changing to 40 btw but can't really spot the difference. Below 30ms I can repro the unwanted flash so I'd go 40-50ms

**Comment by Jacalz** [comment]
It is getting late in Sweden so I will  veck this one out early tomorrow 🙂

I was thinking of names and maby it could be called “Classic” it something because it is the old style 😅

File: js/constants/appConfig.js [hunk 2]
```diff
@@ -153,7 +153,7 @@ module.exports = {
     'tabs.tabs-per-page': 20,
     'tabs.close-action': 'parent',
     'tabs.show-tab-previews': true,
-    'tabs.preview-timing': 1000,
+    'tabs.preview-timing': tabPreviewTiming.SHORT,
```
Comment (bsclifton) [inline js/constants/appConfig.js]: nice!! Love this 😄

**Comment by petemill** [comment]
master 16740dc
0.21.x 9ef9571f84a391671757c61111fc511841f2d3ba 4b91dfe15be436c9ec55ba81cc966245bcdf4953
0.20.x 9ef9571f84a391671757c61111fc511841f2d3ba 4b91dfe15be436c9ec55ba81cc966245bcdf4953

**Comment by Jacalz** [comment]
Can’t get the browser to build because I reinstalled my computer 🖥Going to need to wait for a new beta build from @clifton 🙂

**Pull request merged**